File: DynamicTG-common/src/main/java/tg/dtg/events/Attribute.java
package tg.dtg.events;

import tg.dtg.common.values.Value.ValType;

import java.io.Serializable;

public class Attribute implements Serializable {
  public final String name;
  public final ValType valType;

  public Attribute(String name, ValType valType) {
    this.name = name;
    this.valType = valType;
  }
}

File: DynamicTG-spark/src/main/scala/tg/graph/Attribute.scala
package tg.graph

import com.google.common.collect
import tg.dtg.common.values.Value

trait Attribute extends Serializable

case class RangeAttr(range: collect.Range[Value]) extends Attribute

object RangeAttr {
  def apply(min:Value,max:Value): RangeAttr = {
    RangeAttr(collect.Range.closedOpen(min,max))
  }
}

case class Attr(value: Value) extends Attribute
